audit-service/src/main/java/uk/gov/gchq/palisade/service/audit/stream/config/AkkaRunnableGraph.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2020 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.gchq.palisade.service.audit.stream.config;
+
+import akka.Done;
+import akka.kafka.ConsumerMessage.Committable;
+import akka.kafka.ConsumerMessage.CommittableMessage;
+import akka.kafka.javadsl.Consumer;
+import akka.kafka.javadsl.Consumer.Control;
+import akka.kafka.javadsl.Consumer.DrainingControl;
+import akka.stream.ActorAttributes;
+import akka.stream.Materializer;
+import akka.stream.Supervision;
+import akka.stream.Supervision.Directive;
+import akka.stream.javadsl.RunnableGraph;
+import akka.stream.javadsl.Sink;
+import akka.stream.javadsl.Source;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import scala.Function1;
+
+import uk.gov.gchq.palisade.service.audit.model.AuditErrorMessage;
+import uk.gov.gchq.palisade.service.audit.model.AuditSuccessMessage;
+import uk.gov.gchq.palisade.service.audit.model.Token;
+import uk.gov.gchq.palisade.service.audit.service.AuditServiceAsyncProxy;
+import uk.gov.gchq.palisade.service.audit.service.KafkaProducerService;
+import uk.gov.gchq.palisade.service.audit.stream.ConsumerTopicConfiguration;
+
+import java.nio.charset.Charset;
+import java.util.concurrent.CompletionStage;
+
+/**
+ * Configuration for the Akka Runnable Graph used by the {@link uk.gov.gchq.palisade.service.audit.AuditApplication}
+ * Configures the connection between Kafka, Akka and the service
+ */
+@Configuration
+public class AkkaRunnableGraph {
+    private static final int PARALLELISM = 1;
+
+    @Bean
+    KafkaProducerService kafkaProducerService(final Sink<ProducerRecord<String, AuditErrorMessage>, CompletionStage<Done>> errorSink,
+                                              final Sink<ProducerRecord<String, AuditSuccessMessage>, CompletionStage<Done>> successSink,
+                                              final ConsumerTopicConfiguration upstreamConfig,
+                                              final Materializer materializer) {
+        return new KafkaProducerService(errorSink, successSink, upstreamConfig, materializer);
+    }
+
+    @Bean
+    Function1<Throwable, Directive> supervisor() {
+        return ex -> Supervision.resumingDecider().apply(ex);
+    }
+
+    @Bean
+    RunnableGraph<DrainingControl<Done>> errorRunner(
```

Comment:
Can this method easily be tested and similarly the one for the regular log messages?

Reply:
Both of them are tested in the KafkaContractTest class as each teat scenario will add messages to either the `error` or `success` topics. This is setup in the ContractTestData class.